File: TextosJapones/src/main/java/org/jisho/textosJapones/controller/mangas/MangasTextoCelulaController.java
package org.jisho.textosJapones.controller.mangas;

import com.jfoenix.controls.JFXSlider;
import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.image.Image;
import javafx.scene.image.ImageView;
import javafx.scene.layout.AnchorPane;
import javafx.scene.text.Text;
import org.jisho.textosJapones.components.ImageViewZoom;

import java.net.URL;
import java.util.ResourceBundle;

public class MangasTextoCelulaController implements Initializable {

	@FXML
	public AnchorPane root;

	@FXML
	public ImageView imagem;

	@FXML
	public JFXSlider slider;

	@FXML
	public Text pagina;
	
	public void setDados(Image imagemOriginal, String namePagina) {
		imagem.setImage(imagemOriginal);
		pagina.setText(namePagina);
		
		ImageViewZoom.configura(imagemOriginal, imagem, slider);
	}

	@Override
	public void initialize(URL location, ResourceBundle resources) {
		imagem.fitWidthProperty().bind(root.widthProperty());
		imagem.fitHeightProperty().bind(root.heightProperty());
		imagem.setPreserveRatio(true);
	}

	public static URL getFxmlLocate() {
		return MangasTextoCelulaController.class.getResource("/view/mangas/MangaTextoCelula.fxml");
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/controller/mangas/MangasTextoCelulaController.kt
package br.com.fenix.processatexto.controller.mangas

import br.com.fenix.processatexto.components.ImageViewZoom
import com.jfoenix.controls.JFXSlider
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.image.Image
import javafx.scene.image.ImageView
import javafx.scene.layout.AnchorPane
import javafx.scene.text.Text
import java.net.URL
import java.util.*


class MangasTextoCelulaController : Initializable {
    
    @FXML
    lateinit var root: AnchorPane

    @FXML
    lateinit var imagem: ImageView

    @FXML
    lateinit var slider: JFXSlider

    @FXML
    lateinit var pagina: Text

    fun setDados(imagemOriginal: Image, namePagina: String) {
        imagem.image = imagemOriginal
        pagina.text = namePagina
        ImageViewZoom.configura(imagemOriginal, imagem, slider)
    }

    @Override
    override fun initialize(location: URL, resources: ResourceBundle) {
        imagem.fitWidthProperty().bind(root.widthProperty())
        imagem.fitHeightProperty().bind(root.heightProperty())
        imagem.isPreserveRatio = true
    }

    companion object {
        val fxmlLocate: URL get() = MangasTextoCelulaController::class.java.getResource("/view/mangas/MangaTextoCelula.fxml") as URL
    }
}